tensorbay/opendataset/CCPD/loader.py

```diff
@@ -0,0 +1,229 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name, missing-module-docstring
+
+import os
+from typing import Dict, Iterator, Union
+
+from ...dataset import Data, Dataset
+from ...label import LabeledPolygon
+from .._utility.glob import glob
+
+_PROVINCES = (
+    "皖",
+    "沪",
+    "津",
+    "渝",
+    "冀",
+    "晋",
+    "蒙",
+    "辽",
+    "吉",
+    "黑",
+    "苏",
+    "浙",
+    "京",
+    "闽",
+    "赣",
+    "鲁",
+    "豫",
+    "鄂",
+    "湘",
+    "粤",
+    "桂",
+    "琼",
+    "川",
+    "贵",
+    "云",
+    "藏",
+    "陕",
+    "甘",
+    "青",
+    "宁",
+    "新",
+    "警",
+    "学",
+    "O",
+)
+
+_CODES = (
+    "A",
+    "B",
+    "C",
+    "D",
+    "E",
+    "F",
+    "G",
+    "H",
+    "J",
+    "K",
+    "L",
+    "M",
+    "N",
+    "P",
+    "Q",
+    "R",
+    "S",
+    "T",
+    "U",
+    "V",
+    "W",
+    "X",
+    "Y",
+    "Z",
+    "O",
+)
+
+_ADS = (
+    "A",
+    "B",
+    "C",
+    "D",
+    "E",
+    "F",
+    "G",
+    "H",
+    "J",
+    "K",
+    "L",
+    "M",
+    "N",
+    "P",
+    "Q",
+    "R",
+    "S",
+    "T",
+    "U",
+    "V",
+    "W",
+    "X",
+    "Y",
+    "Z",
+    "0",
+    "1",
+    "2",
+    "3",
+    "4",
+    "5",
+    "6",
+    "7",
+    "8",
+    "9",
+    "O",
+)
+
+_CCPD_SEGMENTS = {
+    "other": ("np", "weather"),
+    "train": ("base",),
+    "val": ("base",),
+    "test": ("blur", "challenge", "db", "fn", "rotate", "tilt"),
+}
+
+DATASET_NAME = "CCPD"
+
+
+def CCPD(path: str) -> Dataset:
+    """Dataloader of CCPD open dataset.
+
+    .. _CCPD: https://github.com/detectRecog/CCPD
+
+    The file structure should be like::
+
+        <path>
```

Comment:
is that necessary to keep the `CCPD2019` layer?
can we adjust the file structure like this:
```
<path>/
    ccpd_np/
        ...
    ccpd_base/
        ...
    ...
```

Reply:
yes,When we download the data, the resulting package is CCPD2019.tar.xz